front-end/src/components/modal/create/VideoPlayer.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef, useState } from 'react';
+import '../../../styles/videoPlayerStyles.css';
+import { useLocation } from 'react-router-dom';
+
+const VideoPlayer = () => {
+  
+  const chartInformation = JSON.parse(decodeURIComponent(new URLSearchParams(useLocation().search).get('chartInformation')));
+    
+  const [videoURL, setVideoURL] = useState('');
+  const videoRef = useRef(null);
+  const [currentTime, setCurrentTime] = useState(0); // added
+  const [duration, setDuration] = useState(0); // scrolly
+
+  useEffect(() => {
+    // Fetch data when the component mounts
+    fetch(`http://${window.location.hostname}:8080/files/${chartInformation.video.key}`)
+      .then((response) => response.blob())
+      .then((blob) => {
+        const url = URL.createObjectURL(blob) 
+        setVideoURL(url);
+
+        return () => {
+          URL.revokeObjectURL(url);
+        };
+      });
+  }, []); // Empty dependency array ensures that the fetch is only performed once
+
+  useEffect(() => {
+    const handleTimeUpdate = () => {
+      setCurrentTime(videoRef.current.currentTime ); // (*1000) Convert to milliseconds 
+    };
+
+    const handleDurationChange = () => {
+      setDuration(videoRef.current.duration);
+    }
+
+    videoRef.current.addEventListener('timeupdate', handleTimeUpdate);
+    videoRef.current.addEventListener('durationchange', handleDurationChange);
+    return () => {
+      videoRef.current.removeEventListener('timeupdate', handleTimeUpdate);
+      videoRef.current.removeEventListener('durationchange', handleDurationChange);
+    };
+  }, []);
```

Comment:
Another duplicate useEffect, which can be done with functions and a single useEffect

Reply:
removed duplicate useEffect